### PR TITLE
Csp 14/feature/linting and ci

### PR DESCRIPTION
**Pull request by alkrauss48**
Addresses #7 and #8

This takes @DeQuasar's work and revamps it to fit our new style of dependencies, as well as a handful of other necessary changes.

This will now add PHP, JS, and Sass linting - all of which will be passing upon a scaffold. Only thing I haven't tested yet is the  github actions file.

@gabe565 I don't mind pushing a scaffold up to a private repo of mine to see if it tests/lints/builds as expected, but wanted to get your opinion before doing that.

It also adds back in the nova page builder plugin, since it works w/ laravel 9 now

### CODE REVIEW

File: cmd/laravel/cmd.go [hunk 1]
```diff
@@ -2,11 +2,12 @@ package laravel
 
 import (
 	"fmt"
+	"os"
+
 	"github.com/AlecAivazis/survey/v2/terminal"
 	"github.com/clevyr/scaffold/internal/appconfig"
 	"github.com/clevyr/scaffold/templates"
 	"github.com/spf13/cobra"
-	"os"
```
Comment (gabe565) [inline cmd/laravel/cmd.go:9]: I think our IDE import sort settings aren't the same. Did you explicitly set yours to split framework packages up from github packages?

File: internal/module/map.go [hunk 2]
```diff
@@ -3,9 +3,10 @@ package module
 import (
 	"errors"
 	"fmt"
-	"gopkg.in/yaml.v3"
 	"sort"
 
+	"gopkg.in/yaml.v3"
+
```
Comment (gabe565) [inline internal/module/map.go:9]: Another IDE import sort issue

File: internal/module/npm.go [hunk 3]
```diff
@@ -65,3 +66,7 @@ func (m NpmMap) InstallDeps() (err error) {
 
 	return
 }
+
+func (NpmMap) RunLintFix() error {
+	return iexec.NewBuilder("npm", "run", "lint:fix").Run()
```
Comment (gabe565) [inline internal/module/npm.go:71]: Any reason to not do this in a `then.run` action? I'm trying to make the actual scaffold be less opinionated and have most of the business logic stored in config/templates

File: templates/laravel-10-before-composer/.github/workflows/laravel.yml [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+          GCLOUD_KEY_FILE: {{`${{ secrets.GCLOUD_KEY_FILE }}`}}
+          GITHUB_TOKEN: {{`${{ secrets.GITHUB_TOKEN }}`}}
+      - name: Post to Slack
+        uses: 8398a7/action-slack@v2
+        if: always()
+        with:
+          status: {{`${{ job.status }}`}}
+          author_name: {{`${{ env.name }}`}} Deploy Status
+        env:
+          SLACK_WEBHOOK_URL: {{`${{ secrets.SLACK_WEBHOOK }}`}}
```
Comment (gabe565) [inline templates/laravel-10-before-composer/.github/workflows/laravel.yml:240]: Please remove the build and deploy step. These will fail before a dev site has been created so they should be left off at first.

I also don't love that the templating required lots of literal strings here. This is fine to merge but I'll make another issue to see if there's a solution to that.

**Review by gabe565 — CHANGES_REQUESTED** [review]
Looks good! Left some comments. Let me know what you think.

File: internal/module/npm.go (hunk 3 above)
Comment (alkrauss48) [reply internal/module/npm.go:71]: I'll put this as a then on the `eslint` task, and if it works then I'll push up the change.

Comment (alkrauss48) [reply internal/module/npm.go]: This worked, and has now been updated

**Review by gabe565 — APPROVED** [review]

**Pull request merged**